src/evict/evict_lru.c

```diff
@@ -1921,8 +1930,10 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
             __wt_cache_read_gen_new(session, page);
 
         /* Pages being forcibly evicted go on the urgent queue. */
-        if (modified &&
-          (page->read_gen == WT_READGEN_OLDEST || page->memory_footprint >= btree->splitmempage)) {
+        if ((modified &&
+              (page->read_gen == WT_READGEN_OLDEST ||
+                page->memory_footprint >= btree->splitmempage)) ||
+          (WT_IS_HS(btree) && __wt_cache_hs_dirty(session))) {
```

Comment:
The condition including history store content deserves a comment - it's not obvious why they are being prioritized.

Reply:
e852c4b